### DISCUSSION TITLE
redirectProxyUrl not possible with PKCE?

### DISCUSSION THREAD

**Discussion by samtgarson** [nextauthjs/next-auth#9347] [Help]
PKCE functionality seems to leave a cookie (`__Secure-next-auth.pkce.code_verifier`) which the app relies on to verify the PKCE challenge on the other side of the OAuth redirect.

However, when using the `redirectProxyUrl`/`AUTH_REDIRECT_PROXY_URL` functionality, it seems that the PKCE challenge occurs on the stable proxy domain, not the preview domain which is being authenticated?

As the cookie above is created with the preview domain (not the stable proxy domain), the PKCE code verifier is not accessible on the stable domain, and I'm getting an error:
```
validCheck: PKCE code_verifier cookie was missing. 
```

Am I setting something up incorrectly or is this a bug?

FWIW I'm authenticating with Google.

**Comment by samtgarson** [comment]
Seems like this is some kind of configuration issue, unable to recreate on an isolated recreation

**Reply by musabash** [reply]
Hi, I am getting the same error after I configured everything as explained [here](https://authjs.dev/getting-started/deployment#securing-a-preview-deployment).
Have you managed to find a solution for this one?
